service/controller/resource/capzcrs/create_test.go

```diff
@@ -106,6 +106,9 @@ func Test_AzureConfigCRMapping(t *testing.T) {
 			r := constructCapzcrsHandler(t, client, tc.location)
 
 			err = r.EnsureCreated(context.Background(), azureConfig)
+			if err != nil {
```

Comment:
mm... I believe this should not be here, so you can use the switch/case later and assert expected errors.